pkg/manager/member/tidb_member_manager.go

```diff
@@ -734,6 +748,20 @@ func getNewTiDBSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap)
 	if tc.IsTLSClusterEnabled() {
 		scheme = corev1.URISchemeHTTPS
 	}
+
+	podSpec := baseTiDBSpec.BuildPodSpec()
+	if baseTiDBSpec.HostNetwork() {
+		podSpec.DNSPolicy = corev1.DNSClusterFirstWithHostNet
+		envs = append(envs, corev1.EnvVar{
```

Comment:
If `EnableAdvertiseAddress` enabled but `HostNetwork` is not enabled, we do not need to set this env?

Reply:
Yes. If HostNetwork is not enabled, $HOSTNAME will be $POD_NAME. 
I've updated code, only set NAMESPACE/HEADLESS_SERVICE_NAME/POD_NAME env when enableAdvertiseAddress is enabled.